app/controllers/users_controller.rb

```diff
@@ -0,0 +1,25 @@
+get "/users" do
+  erb :'users/index'
+end
+
+get "/users/new" do
+  erb :'users/new'
+end
+
+get '/users/:id' do
+  puts @user
+  @suggestions = Suggestion.all
+  @user = User.find_by(id: params[:id])
+  redirect '/' unless @user.id == session[:user_id]
```

Comment:
👍  protecting the serving of the user profile. But use your `current_user` helper here, don't reach directly into the session:
```ruby
redirect '/' unless @user == current_user
```